.github/workflows/test.yaml

```diff
@@ -0,0 +1,44 @@
+name: Tests
+
+on:
+  workflow_call:
+  pull_request:
+
+jobs:
+  lint-and-unit-test:
+    name: Lint and unit tests
+    runs-on: ubuntu-20.04
+    outputs:
+      report: ${{ steps.export-test-report.outputs.report }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install dependencies
+        run: python3 -m pip install tox
+      - name: Run tests
+        run: tox -e lint,unit,coverage-report --result-json=test-result.json
```

Comment:
You don't have a `coverage-report` target defined in `tox.ini`. Might be worth adding that now.